### DISCUSSION TITLE
Syntax Suggestions

### DISCUSSION THREAD

**Discussion by OkayItsMikael** [thinknimble/tn-models-fp#117] [Ideas]
Hey team, thanks for all the great work on this library! I've been thinking about some ways to streamline the setup a bit, and I have some ideas about how that could be done. I figured it would be better to open up a discussion to figure out what the best solution is before creating an issue. Let me know your thoughts!

Suggestions for better DX:

- `createApi`: Rename the `models` option key to `modelSchemas` or `schemas`
- `createApi`: Rename `models:entity` option key to `baseFields` or similar
- `createCustomServiceCall`: Callback function is passed as a second param, it should be on the same level as the other options in the first param object
- `createCustomServiceCall`: Can you create a spec for each api call in one customServiceCall object? Instead of needing separate ones for create, delete, retrieve, etc.
- `createCustomServiceCall`: Rename `inputShape` and `outputShape` to `inputFields` and `outputFields`

Here's a sample that we can use to discuss any other changes or improvements to be made:
```
const list = createPaginatedServiceCall({
  outputShape: videoShape,
  filtersShape: {
    hiveTags: z.string().nullable(),
  },
})

export const educationalVideosApi = createApi(
  {
    client: axiosInstance,
    baseUri: 'api/educational-videos/',
    models: {
      entity: videoShape,
      list: videoShape,
    },
  },
  { list },
)
```

I'm curious about the second parameter in the `createApi` function: could this be consolidated into the first option parameter under a key named `apiOverrides` or something similarly semantic?

**Comment by lakardion** [comment]
@OkayItsMikael With v3 we introduced the consolidation of the parameters into a single one!
This was thanks to a TS release earlier this year. There was some inferences issues that were preventing us to do everything in the same parameter.

**Comment by lakardion** [comment]
> createCustomServiceCall: Can you create a spec for each api call in one customServiceCall object? Instead of needing separate ones for create, delete, retrieve, etc.

I don't think I follow this one, can you pseudo code what this would look like for you as a user of the library?
